include/swift/AST/Decl.h

```diff
@@ -5471,6 +5476,7 @@ enum class ObjCSubscriptKind {
 class SubscriptDecl : public GenericContext, public AbstractStorageDecl {
   SourceLoc StaticLoc;
   SourceLoc ArrowLoc;
+  SourceLoc EndLoc;
```

Comment:
Do properties have the same problem? After all, they can also have unparsed bodies. See `PatternBindingDecl::getSourceRange`.

If that's the case, this should probably go into `AbstractStorageDecl::AccessorRecord` instead. You could even share the storage currently used for `SourceRange Braces` by changing it to something like `SourceLoc BraceStartLoc; /* which may be invalid */` and `SourceLoc EndLoc; /* which is always valid */`.

Reply:
Thanks for the thoughts. I am tempted by the idea of moving it up to `AbstractStorageDecls`, although it's not needed right now.  I'm not crazy about reusing the storage in `AccessorRecord`  because the concept of the end of an `AbstractStorageDecl` seems unrelated to me. It feels like it creates brittleness to mix it in there, and the need for validity comments has a bad smell. Given that, I don't favor moving the end loc up the class hierarchy because, being unused for `VarDecls`, nothing would break if the computation were wrong.
And there's the storage penalty in that case. 

So, after thinking about it, I favor leaving the EndLoc where it is. What do you think?

Reply:
I'm not sure what you mean about it being unused for VarDecls. Looking at the code, I don't see how a PatternBindingDecl would have the right source range for a variable with an unparsed inline getter.

Reply:
I'm coming back to this now that ASTScopes seem to work for building Swift.o. @slavapestov suggested landing it with the default turned on just for Swift.o, and I like the idea. But I'll need to land a few small PRs first, including this one. Would you be OK with not pushing this up for now? I'd rather work on landing the ASTScope stuff, and it doesn't seem to need a SourceRange change for VarDecls. I can file a radar to go back and look at it later if you like.

Reply:
All right, that sounds fair!

Reply:
Thanks! I've created the radar. Will await any feedback on the latest two commits.